### PR TITLE
[ADD] bypass monitor header feature

### PR DESCRIPTION
**Pull request by ilaipx**

### CODE REVIEW

File: PerimeterXModule/Internals/PxContext.cs [hunk 1]
```diff
@@ -209,11 +210,14 @@ private bool shouldMonitorRequest(String uri, PxModuleConfigurationSection pxCon
 				}
 				return true;
 			}
-			else
-			{
-				return pxConfiguration.MonitorMode;
-			} 
-           
+			
+			if (!string.IsNullOrEmpty(pxConfiguration.ByPassMonitorHeader)) {
```
Comment (yaronpx) [inline PerimeterXModule/Internals/PxContext.cs]: the code style convention in C# (unfortunately) is to put the opening brackets in a new line

Comment (ilaipx) [reply PerimeterXModule/Internals/PxContext.cs]: good catch

File: PerimeterXModule/Internals/PxContext.cs [hunk 2]
```diff
@@ -190,30 +190,36 @@ public PxContext(HttpContext context, PxModuleConfigurationSection pxConfigurati
...
-			}
-			else
-			{
-				return pxConfiguration.MonitorMode;
-			} 
-           
+            var mitigationUrls = pxConfiguration.MitigationUrls;
+            if (mitigationUrls.Count > 0)
+            {
+                if (mitigationUrls.Contains(uri))
```
Comment (yaronpx) [inline PerimeterXModule/Internals/PxContext.cs]: I know it's not your code originally, but lines 207 to 211 can be replaced with `return !mitigationUrls.Contains(uri)`

Comment (ilaipx) [reply PerimeterXModule/Internals/PxContext.cs]: I thought you referred to this party also: `if (mitigationUrls.Count > 0)`. Will fix.

**Pull request merged**